nexus-config/src/postgres_config.rs

```diff
@@ -32,6 +33,29 @@ impl PostgresConfigWithUrl {
     pub fn url(&self) -> String {
         self.url_raw.clone()
     }
+
+    /// Accesses the first ip / port pair within the URL.
+    ///
+    /// # Safety
+    ///
+    /// This method makes the assumption that the hostname has at least one
+    /// "host IP / port" pair which can be extracted. If the supplied URL
+    /// does not have such a pair, this function will panic.
+    // Yes, panicking in the above scenario sucks. But this type is already
+    // pretty ubiquitous within Omicron, and integration with the qorb
+    // connection pooling library requires access to database by SocketAddr.
+    pub fn address(&self) -> SocketAddr {
```

Comment:
I don't love that this method makes this assumption, though this appears to be embedded mostly within tests.

It's necessary within `nexus/db-queries/src/db/pool.rs` to implement `SingleHostResolver`, which is our way of identifying backends as `SocketAddr` types.

I can try to make the "address" of the backend generic through qorb, so we could leave it as a URL?